packages/shared/src/hooks/bookmark/useBookmarkFolderList.ts

```diff
@@ -1,23 +1,56 @@
-import { useQuery } from '@tanstack/react-query';
-import { BookmarkFolder, getBookmarkFolders } from '../../graphql/bookmarks';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import {
+  BookmarkFolder,
+  getBookmarkFolders,
+  type MoveBookmarkProps,
+  moveBookmark as moveBookmarkMutation,
+} from '../../graphql/bookmarks';
 import { generateQueryKey, RequestKey, StaleTime } from '../../lib/query';
 import { useAuthContext } from '../../contexts/AuthContext';
+import { getPostByIdKey } from '../usePostById';
+import { useToastNotification } from '../useToastNotification';
 
 interface UseBookmarkFolderList {
   isPending: boolean;
   folders: Array<BookmarkFolder>;
+  isMoving: boolean;
+  onMoveBookmark: (props: MoveBookmarkProps) => void;
 }
 
 export const useBookmarkFolderList = (): UseBookmarkFolderList => {
+  const client = useQueryClient();
+  const { displayToast } = useToastNotification();
   const { isAuthReady, isLoggedIn } = useAuthContext();
   const { data, isPending } = useQuery({
     queryKey: generateQueryKey(RequestKey.BookmarkFolders),
     queryFn: getBookmarkFolders,
     staleTime: StaleTime.Default,
     enabled: isAuthReady && isLoggedIn,
   });
+  const { mutate: moveBookmark, isPending: isMoving } = useMutation({
+    mutationFn: moveBookmarkMutation,
+    onError: () => {
+      displayToast('❌ Failed to move bookmark');
+    },
+  });
+
+  const onMoveBookmark = useCallback(
+    ({ postId, listId }: MoveBookmarkProps) => {
+      moveBookmark({ postId, listId });
+      client.invalidateQueries({
+        queryKey: generateQueryKey(RequestKey.BookmarkFolders),
+      });
+      client.invalidateQueries({
+        queryKey: getPostByIdKey(postId),
+      });
```

Comment:
Why not we move this to `onSuccess`?

Reply:
@sshanzel Because the call returns an empty response. I guess the alternative way would be to pass the postId as prop to the hook, but since we already have it in this method it felt like the better solution

Reply:
@AmarTrebinjac you can access variables passed to mutation fn in [onSuccess](https://tanstack.com/query/latest/docs/framework/react/reference/useMutation#:~:text=back%20optimistic%20updates.-,onSuccess,-%3A%20(data%3A%20TData%2C%20variables)

Reply:
Also I think the second parameter is the variables passed to the function.